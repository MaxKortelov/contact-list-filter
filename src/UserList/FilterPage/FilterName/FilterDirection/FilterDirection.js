import React, {useState} from 'react';
import {Button, Menu, MenuItem} from '@material-ui/core';
import styles from './FilterDirection.module.css';

function FilterDirection({onChangeVector}) {

    const [anchorEl, setAnchorEl] = useState(null);
    const [name, setName] = useState('A - Z');

    const handleClick = (event) => setAnchorEl(event.currentTarget);
    const handleClose = (el) => {
        if(typeof el !== 'object') {
            el === 'A - Z' ? onChangeVector(true) : onChangeVector(false);
            setName(el);
        }
        setAnchorEl(null);
    };

    return(
        <div className={styles.element}>
            <Button variant="contained"
                    color="primary"
                    onClick={handleClick}
            >
                {name}
            </Button>
            <Menu
                open={Boolean(anchorEl)}
                keepMounted
                anchorEl={anchorEl}
                onClose={handleClose}
            >
                <MenuItem onClick={() => handleClose('A - Z')}
                          className={styles.aZ}
                >A - Z</MenuItem>
                <MenuItem onClick={() => handleClose('Z - A')}
                          className={styles.zA}
                >Z - A</MenuItem>
            </Menu>
        </div>
    )
}

export default FilterDirection;
