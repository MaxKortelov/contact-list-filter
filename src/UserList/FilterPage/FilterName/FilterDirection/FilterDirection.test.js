import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import FilterDirection from './FilterDirection';

configure({adapter: new Adapter()});

describe('render FilterDirection', () => {

    const wrapper = shallow(<FilterDirection />);

    it('renders successfully', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
