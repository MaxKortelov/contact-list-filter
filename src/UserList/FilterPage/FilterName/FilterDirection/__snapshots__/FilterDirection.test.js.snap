// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render FilterDirection renders successfully 1`] = `ShallowWrapper {}`;
